File: 1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.cpp
class Solution {
public:
    int isPrefixOfWord(string s, string sw) {
        int m = sw.size();
        int res = 0;
        
        stringstream ss(s);
        string word;

        while (ss >> word){
            res ++;
            if (word.size() >= m && word.substr(0, m) == sw){
                return res;
            }
        }

        return -1;

    }
};

File: 1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.py
class Solution:
    def isPrefixOfWord(self, s: str,sw: str) -> int:
        words = s.split()
        n = len(words)
        m = len(sw)

        for i in range(n):

            if len(words[i]) >= m:
                cnt = 0
                j = 0
                while j < m:
                    if words[i][j] == sw[cnt]:
                        cnt += 1
                    else:
                        break
                    j += 1
                
                if cnt == m:
                    return i + 1
                

        return -1